test/test_models.py

```diff
@@ -51,19 +51,26 @@ def _get_image(input_shape, real_image, device):
 
         img = Image.open(GRACE_HOPPER)
 
-        original_width, original_height = img.size
+        if weights is None:
+            original_width, original_height = img.size
+            # make the image square
+            img = img.crop((0, 0, original_width, original_width))
+            img = img.resize(input_shape[-2:])
 
-        # make the image square
-        img = img.crop((0, 0, original_width, original_width))
-        img = img.resize(input_shape[1:3])
-
-        convert_tensor = transforms.ToTensor()
-        image = convert_tensor(img)
+            convert_tensor = transforms.ToTensor()
+            image = convert_tensor(img)
+        else:
+            H, W = input_shape[-2:]
+            min_side = min(H, W)
+            preprocess = weights.transforms(resize_size=min_side, crop_size=min_side)
```

Comment:
we don't need to pass parameters to the `weights.transforms()` ,  they will handle the size properly.

Reply:
We need this if we want to control the size when the test happened, otherwise we will rely on the default size on the weight transforms (In some big model, we would like to use smaller image size for the test to speed up runtime)